### ISSUE TITLE
Create checkout service interface

### ISSUE THREAD

**Issue by mkucmus** [DivanteLtd/shopware-pwa#73]
Checkout service should be using a Connection layer (based on axios library).
Each method of the service will contain specific types of parameters and return typed data, described in this issue on the bottom.

_Custom cart Interfaces listed at the bottom of the issue._

_It's assumed that the current context`s token is provided on request layer._

_consistent error response should be described in separated issue and used globally_

### The list of proposed methods:
```
1. createOrder(ExtraTokenParam?): Order # an Order interface is taken from #54 
2. createGuestOrder(CreateGuestOrderParam): Order # taken from #54 
3. getDeepLinkOrder(GetDeeplinkParam): Order # taken from #54 
4. payOrder(PayOrderParam): PayOrderResponse | HTTP_NO_CONTENT # described below
```

**All the parameters should be converted into query string (or payload) just before sending a request.**
Checkout Params converter was described here: #TODO

### Available parameters:
```
interface ExtraTokenParam {
  token?: string
  name? : string
}
```

```
interface CreateGuestOrderParam { # some parts of Customer data model's interface. not sure if every key can be used
  email: string
  salutationId: string | uuid
  firstName: string,
  lastName: string,
  billingAddress: BillingAddress
  shippingAddress?: ShippingAddress # shippingAddress interface is taken from #54 
}

```

```
interface GetDeeplinkParam {
  orderId: string | uuid
  accessCode: string
}
```

```
interface PayOrderParam {
  orderId: string | uuid
  finishUrl: string
}
```

### The list of Checkout custom interfaces:
```
interface PayOrderResponse {
 paymentUrl: string
}
```

### Method to endpoint map
```
1. createOrder() - POST /sales-channel-api/v{version}/checkout/order
2. createGuestOrder) - POST /sales-channel-api/v{version}/checkout/guest-order"
3. getDeepLinkOrder() - GET /sales-channel-api/v{version}/checkout/guest-order/{id}
4. payOrder()  - POST /sales-channel-api/v{version}/checkout/order/{orderId}/pay
```

**Comment by mkucmus** [comment]
points 1 and 2 are solved within #357

**Comment by rmakara** [comment]
Closing the issue. `createOrder()` and `createGuestOrder` have been implemented.

Guest orders and payments will be implemented in separate tasks.

Associated:
* #375 
* https://github.com/DivanteLtd/shopware-pwa/issues/481
* https://github.com/DivanteLtd/shopware-pwa/issues/482
